core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -1240,31 +1240,15 @@ object SnappyContext extends Logging {
    * @param sc
    * @return
    */
-  def getOrCreate(sc: SparkContext): SnappyContext = {
-    val gnc = _anySNContext
-    if (gnc != null) gnc
-    else contextLock.synchronized {
-      val gnc = _anySNContext
-      if (gnc != null) gnc
-      else {
-        apply(sc)
-      }
-    }
-  }
-
-  /**
-   * Returns an existing SnappyContext or create one if does not exists
-   * @param jsc
-   * @return SnappyContext
-   */
-  def getOrCreate(jsc: JavaSparkContext): SnappyContext = {
+  def apply(jsc: JavaSparkContext): SnappyContext = {
     if (jsc != null) {
-      getOrCreate(jsc.sc)
+      newSnappyContext(jsc.sc)
```

Comment:
better have `apply(jsc.sc)` in case someone modifies apply behaviour later (and will almost surely forget updating this java one)